client/web/src/user/settings/auth/ExternalAccountsSignIn.tsx

```diff
@@ -26,15 +25,37 @@ interface GitLabExternalData {
     web_url: string
 }
 
+interface SamlExternalData {
+    username: string
+    Assertions: assertion[]
+}
+
+interface assertion {
+    AttributeStatement: attributeStatement
+}
+
+interface attributeStatement {
+    Attributes: attribute[]
+}
+
+interface attribute {
+    Name: string
+    Values: value[]
+}
+
+interface value {
+    Value: string
+}
+
 export interface NormalizedMinAccount {
     name: string
     icon: React.ComponentType<React.PropsWithChildren<{ className?: string }>>
     // some data may be missing if account is not setup
     external?: {
         id: string
         userName: string
-        userLogin: string
-        userUrl: string
+        userLogin: string | null
+        userUrl: string | null
```

Comment:
```suggestion
        userLogin?: string
        userUrl?: string
```

I'm not 100% sure but I think this would be more the convention in our codebase.

Reply:
+1